src/main/java/Scheduler.java

```diff
@@ -22,43 +24,121 @@ public Scheduler(ArrayList<Course> courses, ArrayList<Schedule> schedules) {
+
+            for (Schedule schedule : savedSchedules) {
+                newSchedules.addAll(extendPermutations(newCourse, schedule));
+            }
+
+            savedSchedules.addAll(newSchedules);
+            return savedSchedules;
+        }
+    }
+
+    /**
+     * Takes a list of courses and outputs a schedule that takes the first lecture section and
+     * first tutorial section in each course. Mainly used for testing purposes.
      *
-     * @param courseCodes an ArrayList of course codes from which a schedule will be generated.
+     * @param courses an ArrayList of courses from which a schedule will be generated.
      */
-    public Schedule createBasicSchedule(ArrayList<String> courseCodes) {
+    public Schedule createBasicSchedule(ArrayList<Course> courses) {
         Schedule schedule = new Schedule();
-        Course newCourse;
-
-        for (String courseCode : courseCodes) {
-            try {
-                /**
-                 * For every course code, generate the course from CourseCreator, add first lec/tut
-                 * session to the lectures and tutorials within the schedule.
-                 */
-                newCourse = CourseCreator.generateCourse(courseCode, 'F');
-                System.out.println(newCourse);
-                if (!newCourse.getLectures().isEmpty()) {
-                    schedule.addLecture(newCourse.getLectures().get(0));
+
+        for (Course newCourse : courses) {
+            if (!newCourse.getLectures().isEmpty()) {
+                schedule.addLecture(newCourse.getLectures().get(0));
+            }
+            if (!newCourse.getTutorials().isEmpty()) {
+                schedule.addTutorial(newCourse.getTutorials().get(0));
+            }
+        }
+        return schedule;
+    }
+
+    /** Creates all lecture/tutorial section permutations that pass all filters for one course.
+     *
+     * @param c course that we take lec/tut permutations of
+     * @return all filtered schedules with these lec/tut permutations
+     */
+    private List<Schedule> populatePermutations(Course c) {
+        List<Schedule> populatedSchedules = new ArrayList<>();
+        List<Section> courseLectures = c.getLectures();
+        List<Section> courseTutorials = c.getTutorials();
+
+        for (Section lec : courseLectures) {
+            for(Section tut : courseTutorials) {
+                Schedule tempSchedule = new Schedule();
+                tempSchedule.addLecture(lec);
+                tempSchedule.addTutorial(tut);
+
+                if (this.checkFilters(tempSchedule)) {
+                    populatedSchedules.add(tempSchedule);
                 }
-                if (!newCourse.getTutorials().isEmpty()) {
-                    schedule.addTutorial(newCourse.getTutorials().get(0));
+            }
+        }
+
+        return populatedSchedules;
+    }
+
+    /** Adds creates all permutations of lectures and tutorials that pass filters from a schedule s.
```

Comment:
think u got a typo here

Reply:
also maybe use word besides "pass" because that can also be used to refer to "passing in arguments".

Reply:
I'd change comment to be "Returns all permutations of the Schedule s with the addition of Course c". Idk. Think current comment doesn't make it clear that we're getting permutations of the schedule with a course added